contracts/Marketplace.sol

```diff
@@ -1,51 +1,324 @@
+    function getListingFee() public view returns (uint256) {
+        return listingFee;
+    }
+
+    // * CREATE A LISTING / NFT ITEM
+    // * Create a MarketItem for NFT sale on the Marketplace => Listing
+    // msg.value => Contains the amount of ETH that was sent in the transaction.
+    // msg.sender => Contains the address of the person (or smart contract) who called the current function.
+    function createMarketItem(
+        address nftContract,
+        uint256 tokenId,
+        uint256 price
+    ) public payable nonReentrant {
+        if (price < 0) {
+            revert PriceMoreThanOneWei(); // Price must be greater than 1 wei
+        }
+        if (msg.value != price) {
+            revert FeeEqualToListingFee(); // Fee must be equal to the listing fee
+        }
+        // require(price > 0, "Price must be greater than 1 wei");
+        // require(
+        //     msg.value == listingFee,
+        //     "Fee must be equal to the listing fee"
+        // );
+
+        // When creating the marketItem, we increment the _itemCounter variable by 1 and also, we save that _itemCounter state value to the MarketItem id
+        _itemCounter.increment();
+        uint256 id = _itemCounter.current();
+
+        // We add to the marketItems mapping the new MarketItem with the id as key (generated before).
+        // It contains the data that we want to store for that MarketItem also, the address of the person who called the function and his address.
+        marketItems[id] = MarketItem(
+            id,
+            nftContract,
+            tokenId,
+            payable(msg.sender),
+            payable(address(0)),
+            price,
+            MarketItemState.Listed
+        );
+
+        // * This require seems to be important - Don't know how to implement it.
+        // * We can also pass it a custom error like before, and not using string for it.
+        // ! Modify this require to match ERC1155 contract
+        // require(IERC721(nftContract).getApproved(tokenId) == address(this), "NFT must be approved to market");
+
+        // * Now that we have a MarketItem, we can emit an event with the data that we want to show in the FE?
+        emit MarketItemListed(
+            id,
+            nftContract,
+            tokenId,
+            msg.sender,
+            address(0),
+            price,
+            MarketItemState.Listed
+        );
+    }
+
+    // * DELETE A LISTING
+    // * Delete a MarketItem from the marketplace
+    // ? There is a require statement to check if the owner is the same as the one who made the listing. I'm using the onlyOwner on this function but don't know if works like that.
+    function deleteMarketItem(uint256 itemId) public onlyOwner nonReentrant {
+        if (itemId >= _itemCounter.current()) {
+            revert IdLessOrEqualItemCount();
+        }
+        if (marketItems[itemId].state != MarketItemState.Listed) {
+            revert ItemMustBeOnMarketplace();
+        }
```

Comment:
This might not be needed, depending on what this method ultimately does. Is it a big deal if an item's state is set to `Listed` when it's already `Listed`? The only problem I can see is an extra log being emitted.